gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/PolygonDrawingViewModel.java

```diff
@@ -262,19 +263,48 @@ private boolean isLocationLockEnabled() {
   public void startDrawingFlow(Project selectedProject, Layer selectedLayer) {
     this.selectedLayer.onNext(selectedLayer);
     this.selectedProject.onNext(selectedProject);
-    updateDrawingState(PolygonDrawing.STARTED);
+    polygonDrawingState.onNext(createDrawingState(State.IN_PROGRESS));
+
+    mapPolygon =
+        Optional.of(
+            MapPolygon.newBuilder()
+                .setId(uuidGenerator.generateUuid())
+                .setVertices(ImmutableList.of())
+                .setStyle(selectedLayer.getDefaultStyle())
+                .build());
   }
 
   public LiveData<Integer> getIconTint() {
     return iconTint;
   }
 
-  public LiveData<PolygonFeature> getPolygonFeature() {
-    return drawnPolylineVertices;
+  public LiveData<ImmutableSet<MapFeature>> getMapFeatures() {
+    return drawnMapFeatures;
+  }
+
+  private PolygonDrawingState createDrawingState(State state) {
```

Comment:
This should probably be a `static create()` method in `PolygonDrawingState`

Reply:
Done